pendulum-chain/spacewalk
# replace VecDequeue with a struct _containing_ VecDequeue

potentially closes #309 

Introduce a size limit for maps that stores the transactionevenlope and transactionset.

### General overview of the changes:

1.  Create the `LimitedFifoMap` struct.
2. Changed the datatype for `EnvelopesMap` and `TxSetMap` to a `LimitedFifoMap`
3. Create a test case to for all the methods of `LimitedFifoMap`
4. There were problems connecting to sdftest1, so I've opted to use sdftest2 instead.
------

Things left to do:
- [ ] compare the memory size of the running vault  with the code changes vs the code in main branch 

----

### How to begin the review:
1. Go to vault/src/oracle/types.rs:
   - ```rust
      #[derive(Debug, Clone)]
      pub struct LimitedFifoMap<K, T> {
	      limit: usize,
	      queue: VecDeque<(K, T)>,
      }
      ```
      - check all methods of struct `LimitedFifoMap`, _mimicking_ the method names of the map, except for `first()` and `append()`
      - check the test case `test_LimitedFifoMap` 
2. Go to vault/src/oracle/storage/impl.rs
    - instead of calling 2 methods`push_back()` and  `pop_front()`, the logic is now found in 1 method: `insert()`
3. For other files: 
    - `get_with_key()` is already `get()`   
    - `set_with_key()` is `insert()`
    -  `remove_with_key()` is `remove()`
    -  `contains_key()` is `contains()`
